lib/deploy/lifecycleHooks.js

```diff
@@ -1,43 +1,78 @@
 'use strict';
 
+var _ = require('lodash');
 var RSVP = require('rsvp');
 var utils = require('../utils');
 var chalk = require('chalk');
-var process = require('child_process');
+var cprocess = require('child_process');
 var FirebaseError = require('../error');
+var getProjectId = require('../getProjectId');
+var logger = require('../logger');
+
+
+function runCommand(command, options) {
+  return new Promise(function(resolve, reject) {
+    var child = cprocess.spawn(command, _.assign({}, options, {
+      shell: true,
+      stdio: [0, 1, 2] // Inherit STDIN, STDOUT, and STDERR
+    })
+    );
+    logger.info('Running command: ' + command);
+    console.log(process.env);
+    child.on('error', function(err) {
+      reject(new Error(err));
+    });
+    child.on('exit', function(code, signal) {
+      if (signal !== null) {
```

Comment:
nit: It's a little odd that we're using explicit value checks for signal and truthy checks for code.